maggy/core/experimentdriver.py

```diff
@@ -24,98 +28,185 @@
 class ExperimentDriver(object):
 
     SECRET_BYTES = 8
+    # TODO rewrite with @classmethod
+
+    # @Moritz:
+    # for now, we infer the experiment type (an optimization experiment or an ablation study)
+    # using keyword arguments, and set self.experiment_type to an according string.
+    # Some of these arguments are required for any maggy experiment:
+    # num_trials, name, num_executors, hb_interval, description, app_dir, log_dir, trial_dir
+    # while some are specific to the type of experiment. For example, if the ExperimentDriver constructor
+    # is called with a `searchspace` parameter, we infer that it is a hyperparameter optimization task,
+    # and if it is called with `ablationstudy` parameter, we infer that it's an ablation study.
 
-    def __init__(self, searchspace, optimizer, direction, num_trials, name, num_executors, hb_interval, es_policy, es_interval, es_min, description, app_dir, log_dir, trial_dir):
+    def __init__(self, experiment_type, **kwargs):
 
         global driver_secret
 
+        # COMMON EXPERIMENT SETUP
         self._final_store = []
+        self._trial_store = {}
+        self.num_executors = kwargs.get('num_executors')
+        self._message_q = queue.Queue()
+        self.name = kwargs.get('name')
+        self.experiment_done = False
+        self.worker_done = False
+        self.hb_interval = kwargs.get('hb_interval')
+        self.description = kwargs.get('description')
+        self.experiment_type = experiment_type
+        self.es_interval = integer_max  # XXX not the cleanest way
```

Comment:
Could we also set default `es_policy` to `None` in `lagom()`?
Or in general not sure we need it and you can just set it both to the default, if user doesn't provide custom values because of line 150.
```suggestion
        self.es_interval = kwargs.get(es_interval)		
        self.es_min = kwargs.get(es_min)
```

You could also set it to None in case of ablation and add an if statement before line 296.

Reply:
I added `if self.earlystop_check == NoStoppingRule.earlystop_check: ` before `if (datetime.now() - time_earlystop_check).total_seconds() >= self.es_interval: `. This should work since for ablation I already set its `earlystop_check` to `NoStoppingRule.earlystop_check`, but you should double check this again.

Reply:
https://github.com/logicalclocks/maggy/pull/27/commits/2b6977cebde9eb67bbd09db082a31904951f2591#diff-bfdfef7fbe9209260b7ba10581f534edR289